v2/internal/controllers/to_azure_secrets_test.go

```diff
@@ -79,10 +84,131 @@ func Test_MissingSecretKey_ReturnsError(t *testing.T) {
+	secretInDiffNamespace := createNamespacedSecret(tc, nameSpaceName, secretName)
+
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	vnet := newVMVirtualNetwork(tc, testcommon.AsOwner(rg))
+	subnet := newVMSubnet(tc, testcommon.AsOwner(vnet))
+	networkInterface := newVMNetworkInterface(tc, testcommon.AsOwner(rg), subnet)
+
+	tc.CreateResourcesAndWait(vnet, subnet, networkInterface)
+
+	vm := newVirtualMachine20201201(tc, rg, networkInterface, secretInDiffNamespace)
+
+	tc.CreateResourceAndWaitForState(vm, metav1.ConditionFalse, conditions.ConditionSeverityWarning)
+	// We expect the ready condition to include details of the error
+	tc.Expect(vm.Status.Conditions[0].Reason).To(Equal(conditions.ReasonSecretNotFound))
+	tc.Expect(vm.Status.Conditions[0].Message).To(
+		ContainSubstring("failed resolving secret references: %s/%s does not exist", tc.Namespace, secretInDiffNamespace.Name))
+
+	// Delete VM and resources.
+	tc.DeleteResourcesAndWait(vm, networkInterface, subnet, vnet, rg)
+}
+
+func Test_UserSecretInDifferentNamespace_ShouldNotTriggerReconcile(t *testing.T) {
+	t.Parallel()
+	ns1 := "secret-in-different-namespace-ns-1"
+	ns2 := "secret-in-different-namespace-ns-2"
+
+	cfg, err := testcommon.ReadFromEnvironmentForTest()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// TargetNamespaces is only used here for logger mapping in namespaces, and not multi-tenancy.
+	cfg.TargetNamespaces = []string{ns1, ns2}
+	tc := globalTestContext.ForTestWithConfig(t, cfg)
+
+	tc.CreateTestNamespaces(ns1, ns2)
+
+	secretName := tc.Namer.GenerateName("vm-secret")
+	ns1Secret := createNamespacedSecret(tc, ns1, secretName)
+	ns2Secret := createNamespacedSecret(tc, ns2, secretName)
+
+	rg := tc.NewTestResourceGroup()
+	rg.Namespace = ns1
+	tc.CreateResourceGroupAndWait(rg)
+
+	rg2 := tc.NewTestResourceGroup()
+	rg2.Namespace = ns2
+	tc.CreateResourceGroupAndWait(rg2)
+
+	// VM1 with same secret name in ns1
+	vm1 := createNamespacedVM(tc, rg, ns1, ns1Secret)
+
+	resourceID := genruntime.GetResourceIDOrDefault(vm1)
+	tc.Expect(resourceID).ToNot(BeEmpty())
+
+	// Deleting server1 here using AzureClient so that Operator does not know about the deletion.
+	// If operator reconciles that resource again it will be recreated and we will notice
+	_, err = tc.AzureClient.DeleteByID(tc.Ctx, resourceID, vm1.GetAPIVersion())
+	if err != nil {
+		return
+	}
+
+	// VM2 with same secret name in ns2
+	_ = createNamespacedVM(tc, rg2, ns2, ns2Secret)
+
+	// Here we want to make sure that the server1 we deleted from Azure was not created again by a reconcile triggered when the second secret was created
+	// in other namespace
+	_, err = tc.AzureClient.GetByID(tc.Ctx, resourceID, vm1.GetAPIVersion(), vm1)
+	tc.Expect(err).To(HaveOccurred())
+	tc.Expect(genericarmclient.IsNotFoundError(err)).To(BeTrue())
+
+	tc.GetResource(client.ObjectKeyFromObject(vm1), vm1)
+	tc.Expect(vm1.Status.Conditions[0].Type).To(BeEquivalentTo(conditions.ConditionTypeReady))
```

Comment:
You also need to assert that `Conditions` has at least 1 item (or else `[0]` access here will panic).
Also need to assert that the status of the condition is `True`. The `Type` of the condition will always be ready, but it might be `ready = false`, which would indicate a failure of the test.